packages/repository-json-schema/src/filter-json-schema.ts

```diff
@@ -22,6 +25,10 @@ const scopeFilter = getFilterJsonSchemaFor(EmptyModel);
  */
 export function getFilterJsonSchemaFor(modelCtor: typeof Model): JsonSchema {
   const schema: JsonSchema = {
+    title:
+      modelCtor.modelName === 'EmptyModel'
```

Comment:
Checking the model name is not safe enough, because users can have their own model that's called `EmptyModel`. We should compare the model ctor function directly.

```suggestion
      modelCtor === EmptyModel
```

To be honest, I am not very happy that we have to pollute schema-generating functions with checks for EmptyModel. I am concerned that it's too easy for future contributors to forget to implement this check and accidentally introduce `EmptyModel` title into the result.

Since `EmptyModel` was introduced only as a tool to simplify implementation of `scopeFilter`, I prefer to keep all EmptyModel-related handling inside the code building `scopeFilter` value.

How about introducing an optional `options` argument to all `get*SchemaFor` methods, one that will allow the caller to request a schema with no title set?

```ts
const scopeFilter = {
  ...getFilterJsonSchemaFor(EmptyModel, {setTitle: false}),
  title: 'ScopeFilter', // TODO(dougal83) base title on model
};
```

Implementation:

```ts
export function getFilterJsonSchemaFor(
  modelCtor: typeof Model,
  options: FilterSchemaOptions
): JsonSchema {
  const schema: JsonSchema = {
    title: /*...*/,
    properties: {
      // Forward the options. If the caller does not want title for the top-level schema,
      // then we should not set titles in the nested schemas either
      where: getWhereJsonSchemaFor(modelCtor, options),

      fields: getFieldsJsonSchemaFor(modelCtor, options),
   
      // ...
    },
    // ...
  };
  // ...
}
```

Don't forget to write new tests to verify correctness of the implementation handling the new option.

Reply:
Using options is a much better idea!  Will update.